src/CmdLine.hs

```diff
@@ -217,12 +217,11 @@ mode = cmdArgsMode $ modes
 --   We want more important hints to go last, since they override
 cmdHintFiles :: Cmd -> IO [FilePath]
 cmdHintFiles cmd = do
-    let explicit1 = [cmdDataDir cmd </> "hlint.yaml" | null $ cmdWithHints cmd]
```

Comment:
If `cmdWithHints` is specified then we don't use `cmdDataDir` for the hints. It seems in your patch that changes?

Reply:
My bad, should be fixed now